### PR TITLE
darwin: add event ApplicationShouldHandleReopen

### PR DESCRIPTION
**Pull request by 5aaee9**
# Description

This pull request add new Event [ApplicationShouldHandleReopen](https://developer.apple.com/documentation/appkit/nsapplicationdelegate/1428638-applicationshouldhandlereopen?language=objc) to handle dock click in macOS.

This is useful to app which use close button to hide itself (Example in `examples/hide-window`).

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
  
Tested in darwin in the example

- [ ] Windows
- [x] macOS
- [ ] Linux
  
## Test Configuration

```
# System
┌─────────────────────────┐
| OS           | MacOS    |
| Version      | 12.6.7   |
| ID           | 21G651   |
| Go Version   | go1.20.7 |
| Platform     | darwin   |
| Architecture | amd64    |
└─────────────────────────┘

# Dependencies
┌────────────────────────────────────────────────────────────────┐
| Dependency                | Package Name | Status    | Version |
| Xcode command line tools  | N/A          | Installed | 2395    |
| Nodejs                    | N/A          | Installed | 20.7.0  |
| npm                       | N/A          | Installed | 10.1.0  |
| *Xcode                    | N/A          | Available |         |
| *upx                      | N/A          | Available |         |
| *nsis                     | N/A          | Available |         |
└─────────────────── * - Optional Dependency ────────────────────┘
```

# Checklist:

- [x] I have updated `website/src/pages/changelog.mdx` with details of this PR
- [x] My code follows the general coding style of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by leaanthony** [comment]
Thanks for this! If you run `wails3 task generate:events` from the v3 directory then it will generate the correct IDs 🙏

**Comment by 5aaee9** [comment]
@leaanthony Generated new ID list. It correct now?

File: v3/pkg/application/application_darwin_delegate.m [hunk 1]
```diff
@@ -17,6 +17,14 @@ - (void)themeChanged:(NSNotification *)notification {
         processApplicationEvent(EventApplicationDidChangeTheme, NULL);
     }
 }
+- (BOOL)applicationShouldHandleReopen:(NSNotification *)notification 
+                    hasVisibleWindows:(BOOL)flag {
+    if( hasListeners(EventApplicationShouldHandleReopen) ) {
+        processApplicationEvent(EventApplicationShouldHandleReopen, NULL);
```
Comment (leaanthony) [inline v3/pkg/application/application_darwin_delegate.m:23]: Would it be useful to pass the flag with the event?

Comment (5aaee9) [reply v3/pkg/application/application_darwin_delegate.m]: Updated. The `processApplicationEvent` now accept NSDirectory and will save to the `ApplicationEventContext`

**Review by leaanthony — APPROVED** [review]
This is great. Thank you! 🙏

**Pull request merged**